oak/server/oak_node.cc

```diff
@@ -251,4 +251,10 @@ bool OakNode::WaitOnChannels(std::vector<std::unique_ptr<ChannelStatus>>* status
   }
 }
 
+void OakNode::ClearHandles() {
+  absl::MutexLock lock(&mu_);
+  OAK_LOG(INFO) << "Dropping " << channel_halves_.size() << " handles for Node";
+  channel_halves_.clear();
```

Comment:
I presume this correctly orphans channels as appropriate?

Reply:
Yeah, the `channel_halves_` is a map holding `ChannelHalf` object, which in turn own either a `MessageChannelWriteHalf` or a `MessageChannelReadHalf`, either of which has a `channel_` field that's a `shared_ptr` to a `MessageChannel`.  So the refcount will be dropped as a result, orphaning as appropriate.